test/api_integration/apis/search/bsearch.ts

```diff
@@ -0,0 +1,166 @@
+                params: {
+                  body: {
+                    query: {
+                      match_all: {},
+                    },
+                  },
+                },
+              },
+            },
+          ],
+        });
+
+        expect(resp.status).to.be(200);
+        const parsedResponse = parseBfetchResponse(resp);
+        expect(parsedResponse).to.have.length(2);
+        parsedResponse.forEach((responseJson) => {
+          expect(responseJson.result.isPartial).to.be(false);
+          expect(responseJson.result.isRunning).to.be(false);
+          expect(responseJson.result).to.have.property('rawResponse');
+        });
+      });
+
+      it('should return error for not found strategy', async () => {
+        const resp = await supertest.post(`/internal/bsearch`).send({
+          batch: [
+            {
+              request: {
+                params: {
+                  body: {
+                    query: {
+                      match_all: {},
+                    },
+                  },
+                },
+              },
+              options: {
+                strategy: 'wtf',
+              },
+            },
+          ],
+        });
+
+        expect(resp.status).to.be(200);
+        parseBfetchResponse(resp).forEach((responseJson, i) => {
+          expect(responseJson.id).to.be(i);
+          verifyErrorResponse(responseJson.error, 404, 'Search strategy wtf not found');
+        });
+      });
+
+      it('should return 400 when index type is provided in OSS', async () => {
+        const resp = await supertest.post(`/internal/bsearch`).send({
+          batch: [
+            {
+              request: {
+                indexType: 'baad',
+                params: {
+                  body: {
+                    query: {
+                      match_all: {},
+                    },
+                  },
+                },
+              },
+            },
+          ],
+        });
+
+        expect(resp.status).to.be(200);
+        parseBfetchResponse(resp).forEach((responseJson, i) => {
+          expect(responseJson.id).to.be(i);
+          verifyErrorResponse(responseJson.error, 400, 'Unsupported index pattern type baad');
+        });
+      });
+
+      it('should return 400 for Painless error', async () => {
+        await esArchiver.loadIfNeeded(
+          '../../../functional/fixtures/es_archiver/logstash_functional'
+        );
+        const resp = await supertest.post(`/internal/bsearch`).send({
+          batch: [
+            {
+              request: painlessErrReq,
+            },
+          ],
+        });
+
+        expect(resp.status).to.be(200);
+        parseBfetchResponse(resp).forEach((responseJson, i) => {
+          expect(responseJson.id).to.be(i);
+          verifyErrorResponse(responseJson.error, 400, 'search_phase_execution_exception', true);
+        });
+
+        await esArchiver.unload('../../../functional/fixtures/es_archiver/logstash_functional');
```

Comment:
unload should happen in `after` hook, because otherwise if the test fails midway - it doesn't unload